runtime/src/main/java/com/datastax/oss/quarkus/runtime/internal/quarkus/CassandraClientProducer.java

```diff
@@ -149,6 +149,12 @@ private void configureConnectionSettings(
         DefaultDriverOption.LOAD_BALANCING_LOCAL_DATACENTER, config.localDatacenter);
     config.requestTimeout.ifPresent(
         v -> configLoaderBuilder.withDuration(DefaultDriverOption.REQUEST_TIMEOUT, v));
+    config.username.ifPresent(
+        username ->
+            configLoaderBuilder.withString(DefaultDriverOption.AUTH_PROVIDER_USER_NAME, username));
+    config.password.ifPresent(
+        password ->
+            configLoaderBuilder.withString(DefaultDriverOption.AUTH_PROVIDER_PASSWORD, password));
```

Comment:
If you don't specify the AuthProvider class here, the driver won't create any for you.

```suggestion
    if (config.username.isPresent() && config.password.isPresent()) {
      configLoaderBuilder
          .withClass(DefaultDriverOption.AUTH_PROVIDER_CLASS, PlainTextAuthProvider.class)
          .withString(DefaultDriverOption.AUTH_PROVIDER_USER_NAME, config.username.get())
          .withString(DefaultDriverOption.AUTH_PROVIDER_PASSWORD, config.password.get());
    }
```